### ISSUE TITLE
AWS like search bar directive (1st iteration)

### ISSUE THREAD

**Issue by jesusgn90** [wazuh/wazuh-kibana-app#1002]
Hi team, this is a gif from the AWS EC2 search bar:

![ec2-search-bar](https://user-images.githubusercontent.com/8860227/47663555-991d9000-db9d-11e8-805d-701040118794.gif)

It would be nice to have a similar search bar for our app. This task is divided into the next points:

### Iteration 1 (49th week sprint)

- [x] It must be an Angular.js directive (reusable, parameterizable)
- [x] It must accept a parameter where we define the suggestions that the search bar will show
- [x] Handle empty searches properly
- [x] When adding filters it should work as follow: if there is a similar filter, the new filter should be an OR operation with the existing ones. If there is no similar filter, the new filter must be an AND operation with the existing ones. Example: `(group: default OR group: staging) AND os: Linux`
```
GET "/agents?search=win&q=(group=staging,group=default);os.platform=windows"
```

- [x] Filters must be removable.
- [x] Remove all the current selectors. Example: the agents list has some selectors, their logic should be supplied by this new directive.
- [x] Agents list section

https://github.com/wazuh/wazuh-kibana-app/pull/1058

Regards

**Comment by jesusgn90** [comment]
Blocked by https://github.com/wazuh/wazuh-api/issues/248
